aws/ec2_unit_test.go

```diff
@@ -0,0 +1,284 @@
+// These tests use GoMock and the ec2iface to provide a mock framework for testing the EC2 API
+// Unlike other tests in cloud-nuke, nuking the default VPCs and security groups is not an option.
+// Other tests within cloud-nuke depend on the default VPCs/SGs to function, and other projects
+// may be using the same AWS account at the same time. Deleting the default VPCs would break things.
+// Therefore, the default VPC/SG nuke testing is mocked as unit tests.
+// To generate the EC2API mock, install https://github.com/golang/mock, then use the following:
+// mockgen -source vendor/github.com/aws/aws-sdk-go/service/ec2/ec2iface/interface.go -destination aws/mocks/gomock-EC2API.go
```

Comment:
Ah didn't realize that mock file was autogen'd. Good call on including those instructions!

Reply:
You thought I wrote all 20,920 lines?! I'm flattered!

Reply:
Oh no I thought you copied pasted it from the dark web 😄